hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -166,168 +159,33 @@ protected void doRefresh() {
-        throw new AlreadyExistsException(e, "Table already exists: %s.%s", database, tableName);
-
-      } catch (InvalidObjectException e) {
-        throw new ValidationException(e, "Invalid Hive object for %s.%s", database, tableName);
-
-      } catch (CommitFailedException | CommitStateUnknownException e) {
-        throw e;
-
-      } catch (Throwable e) {
-        if (e.getMessage()
-            .contains(
-                "The table has been modified. The parameter value for key '"
-                    + HiveTableOperations.METADATA_LOCATION_PROP
-                    + "' is")) {
-          throw new CommitFailedException(
-              e, "The table %s.%s has been modified concurrently", database, tableName);
-        }
-
-        if (e.getMessage() != null
-            && e.getMessage().contains("Table/View 'HIVE_LOCKS' does not exist")) {
-          throw new RuntimeException(
-              "Failed to acquire locks from metastore because the underlying metastore "
-                  + "table 'HIVE_LOCKS' does not exist. This can occur when using an embedded metastore which does not "
-                  + "support transactions. To fix this use an alternative metastore.",
-              e);
-        }
-
-        LOG.error(
-            "Cannot tell if commit to {}.{} succeeded, attempting to reconnect and check.",
-            database,
-            tableName,
-            e);
-        commitStatus = checkCommitStatus(newMetadataLocation, metadata);
-        switch (commitStatus) {
-          case SUCCESS:
-            break;
-          case FAILURE:
-            throw e;
-          case UNKNOWN:
-            throw new CommitStateUnknownException(e);
-        }
-      }
-    } catch (TException e) {
-      throw new RuntimeException(
-          String.format("Metastore operation failed for %s.%s", database, tableName), e);
-
-    } catch (InterruptedException e) {
-      Thread.currentThread().interrupt();
-      throw new RuntimeException("Interrupted during commit", e);
-
-    } catch (LockException e) {
-      throw new CommitFailedException(e);
-
-    } finally {
-      cleanupMetadataAndUnlock(commitStatus, newMetadataLocation, lock);
-    }
+    String baseMetadataLocation = base != null ? base.metadataFileLocation() : null;
+    commitWithLocking(conf, base, metadata, baseMetadataLocation, newMetadataLocation, fileIO);
 
     LOG.info(
         "Committed to table {} with the new metadata location {}", fullName, newMetadataLocation);
   }
 
-  @VisibleForTesting
-  Table loadHmsTable() throws TException, InterruptedException {
-    try {
-      return metaClients.run(client -> client.getTable(database, tableName));
-    } catch (NoSuchObjectException nte) {
-      LOG.trace("Table not found {}", fullName, nte);
-      return null;
-    }
+  @Override
+  public CommitStatus validateNewLocationAndReturnCommitStatus(
+      BaseMetadata metadata, String newMetadataLocation) {
```

Comment:
I'm not convinced using `BaseMetadata` here is the best approach just to pass properties or the metadata location(s)